engine/src/main/java/com/datatorrent/stram/client/StramAgent.java

```diff
@@ -417,11 +417,25 @@ private StramWebServicesInfo retrieveWebServicesInfo(String appId)
       String appPath = response.getString("appPath");
       String user = response.getString("user");
       JSONObject permissionsInfo = null;
-      try (FSDataInputStream is = fileSystem.open(new Path(appPath, "permissions.json"))) {
-        permissionsInfo = new JSONObject(IOUtils.toString(is));
+      Path permissionsPath = new Path(appPath, "permissions.json");
+      LOG.debug("Checking for permission information in file {}", permissionsPath);
+      try {
+        if (fileSystem.exists(permissionsPath)) {
+          LOG.info("Loading permission information");
+          try (FSDataInputStream is = fileSystem.open(permissionsPath)) {
+            permissionsInfo = new JSONObject(IOUtils.toString(is));
+          }
+        } else {
+          // ignore and log messages if file is not found
+          LOG.info("Permission information not available, file not present");
+        }
       } catch (IOException ex) {
-        // ignore if file is not found
-        LOG.info("Exception in accessing permissions.json", ex);
+        // ignore and log message when unable to read the file
+        if (LOG.isDebugEnabled()) {
```

Comment:
Regarding logging the exception, the traces will be logged continuously, each time an application info is retrieved. I prefer to just have the message as it is not a critical functionality at the info level.

Regrading, the PermissionInfo class, the loading should be outside the class as IMO it should not have the bias to load permissions in a very specific way from hdfs like that, the way it is being done today or in a separate loader if we want to avoid bloat or perceive reuse. I prefer to leave it the way it is.